package/MDAnalysis/coordinates/XYZ.py

```diff
@@ -259,99 +259,85 @@ class XYZReader(base.Reader):
     def __init__(self, filename, **kwargs):
         super(XYZReader, self).__init__(filename, **kwargs)
 
-        # the filename has been parsed to be either be foo.xyz or foo.xyz.bz2 by coordinates::core.py
-        # so the last file extension will tell us if it is bzipped or not
+        # the filename has been parsed to be either be foo.xyz or foo.xyz.bz2 by
+        # coordinates::core.py so the last file extension will tell us if it is
+        # bzipped or not
         root, ext = os.path.splitext(self.filename)
         self.xyzfile = util.anyopen(self.filename, "r")
         self.compression = ext[1:] if ext[1:] != "xyz" else None
+        self._cache = dict()
 
-        # note that, like for xtc and trr files, _n_atoms and _n_frames are used quasi-private variables
-        # to prevent the properties being recalculated
-        # this is because there is no indexing so the way it measures the number of frames is to read the whole file!
-        self._n_atoms = self._read_xyz_natoms(self.filename)
-        self._n_frames = None
-
-        self.ts = self._Timestep(self._n_atoms, **self._ts_kwargs)
-
-        #        Haven't quite figured out where to start with all the self._reopen() etc.
-        #        (Also cannot just use seek() or reset() because that would break with urllib2.urlopen() streams)
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+        # Haven't quite figured out where to start with all the self._reopen()
+        # etc.
+        # (Also cannot just use seek() or reset() because that would break
+        # with urllib2.urlopen() streams)
```

Comment:
I don't think that any of the trajectory readers actually work with non-seekable streams such as the ones from `urllib2.urlopen()` because we are constantly opening and closing them. I probably wrote this when I had the wildly optimistic vision that this might "just work" ;-).